src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```
@@ -184,6 +188,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="--sysroot=&quot;$(SysRoot)&quot;" Condition="'$(SysRoot)' != '' and '$(_IsApplePlatform)' != 'true'" />
       <LinkerArg Include="-isysroot &quot;$(SysRoot)&quot;" Condition="'$(SysRoot)' != '' and '$(_IsApplePlatform)' == 'true'" />
       <LinkerArg Include="--target=$(TargetTriple)" Condition="'$(_targetOS)' != 'osx' and '$(TargetTriple)' != ''" />
+      <LinkerArg Include="-mmacos-version-min=$(AppleMinOSVersion)" Condition="'$(_targetOS)' == 'osx'" />
```

nit: to support this for $otherApplePlatforms, we can define a property after line 73:

```xml
      <_AppleCompilerFlagName Condition="'$(_targetOS)' == 'osx'">macos</_AppleCompilerFlagName>
      <_AppleCompilerFlagName Condition="'$(_targetOS)' == 'maccatalyst'">maccatalyst</_AppleCompilerFlagName>
      <_AppleCompilerFlagName Condition="'$(_targetOS)' == 'ios'">ios</_AppleCompilerFlagName>
      <_AppleCompilerFlagName Condition="'$(_targetOS)' == 'iossimulator'">iphonesimulator</_AppleCompilerFlagName>
      <_AppleCompilerFlagName Condition="'$(_targetOS)' == 'tvos'">tvos</_AppleCompilerFlagName>
      <_AppleCompilerFlagName Condition="'$(_targetOS)' == 'tvossimulator'">tvos-simulator</_AppleSdkName>
```

and change it like:
```suggestion
      <LinkerArg Include="-m$(_AppleCompilerFlagName)-version-min=$(AppleMinOSVersion)" Condition="'$(_targetOS)' == 'osx'" />
```

inspired by:
https://github.com/dotnet/runtime/blob/022fa98a759a296fc5c84dae57b8f7da18d6b375/src/tasks/MobileBuildTasks/Apple/AppleProject.cs#L80

---

Other Apple platforms already achieve this through the `triple` right above. I used the same name for the `AppleMinOSVersion` property.

---

Ah, missed that. Shall can use the target triple for osx as well? ` --target=x86_64-apple-macos12-darwin`, ` --target=arm64-apple-macos11-darwin` etc.

---

It’s also an option. I don’t have strong preference either way.

---

I'll give it a go but it's slightly bigger change.

---

Sounds good. It's just a consistency nit (either -m or passing version in triplet/quadruplet to -target for all Apple platforms alike would be slightly nicer)

---

I tried `arm64-apple-macos11`, `arm64-apple-darwin16` and few other combination like the ones you mentioned, and neither of them had any effect on `LC_BUILD_VERSION`. Now I am questioning whether it works for the iOS-like platforms.

---

I was testing with quardruplet (which we are using for other Apple platforms):
```zsh
% echo "int main(void) { return 0; }" > x.c
% clang --target=x86_64-apple-macos12-darwin x.c             
% vtool -show a.out | grep minos                             
    minos 12.0
% clang --target=aarch64-apple-macos13-darwin x.c
% vtool -show a.out | grep minos                 
    minos 13.0
```

---

Ah, my bad, I was checking on a wrong path (`publish` vs `native`), so even the triplet works.